src/ucp/core/ucp_ep.c

```diff
@@ -1067,6 +1067,12 @@ static void ucp_ep_config_set_rndv_thresh(ucp_worker_t *worker,
     } else {
         rndv_thresh     = context->config.ext.rndv_thresh;
         rndv_nbr_thresh = context->config.ext.rndv_thresh;
+
+        /* adjust max_short if rndv_thresh is set externally */
+        if (config->tag.eager.max_short > 0) {
```

Comment:
why need the >0 check?

Reply:
how about max_short = ucs_min(max_short, (ssize_t)rndv_thresh-1)

Reply:
still, if rndv_thresh=0, we should set max_short to -1 and not 0 to disable it completely

Reply:
yes, ic

Reply:
> still, if rndv_thresh=0, we should set max_short to -1 and not 0 to disable short protocol completely

yep, you're right. currently, we are using short for 0-length messages

Reply:
fixed